src/transformers/models/bridgetower/modeling_bridgetower.py

```diff
@@ -0,0 +1,1006 @@
+        width: int,
+        layers: int,
+        heads: int,
+        attn_mask: torch.Tensor = None,
+        model_type: str = "bridgetower",
+        stop_gradient: bool = False,
+        vit_remove_last: bool = False,
+    ):
+        super().__init__()
+        self.width = width
+        self.layers = layers
+        if vit_remove_last:
+            self.resblocks = nn.Sequential(
+                *[BridgeTowerResidualAttention(width, heads, attn_mask) for _ in range(layers - 1)]
+            )
+        else:
+            self.resblocks = nn.Sequential(
+                *[BridgeTowerResidualAttention(width, heads, attn_mask) for _ in range(layers)]
+            )
+        self.model_type = model_type
+        self.stop_gradient = stop_gradient
+
+    def forward(self, x: torch.Tensor, x_mask: torch.Tensor = None):
+        xs = []
+        for block in self.resblocks:
+            x = block(x, x_mask)
+            if self.model_type == "bridgetower":
+                if self.stop_gradient:
+                    xs.append(x.detach())
+                else:
+                    xs.append(x)
+        if self.model_type == "bridgetower":
+            return xs
+        else:
+            return x
+
+
+class BridgeTowerVisualTransformer(nn.Module):
+    def __init__(
+        self,
+        input_resolution: int,
+        patch_size: int,
+        width: int,
+        layers: int,
+        heads: int,
+        output_dim: int,
+        resolution_after: int,
+        model_type: str = "bridgetower",
+        stop_gradient: bool = False,
+        vit_layernorm_shared: bool = True,
+        vit_remove_last: bool = False,
+    ):
+        super().__init__()
+        self.input_resolution = input_resolution
+        self.output_dim = output_dim
+        self.conv1 = nn.Conv2d(
+            in_channels=3, out_channels=width, kernel_size=patch_size, stride=patch_size, bias=False
+        )
+
+        scale = width**-0.5
+        self.class_embedding = nn.Parameter(scale * torch.randn(width))
+        self.positional_embedding = nn.Parameter(scale * torch.randn((resolution_after // patch_size) ** 2 + 1, width))
+        self.ln_pre = BridgeTowerLayerNorm(width)
+
+        self.transformer = BridgeTowerTransformer(
+            width, layers, heads, model_type=model_type, stop_gradient=stop_gradient, vit_remove_last=vit_remove_last
+        )
+        self.ln_post = BridgeTowerLayerNorm(width)
+        self.model_type = model_type
+        self.vit_layernorm_shared = vit_layernorm_shared
+        if not vit_layernorm_shared:
+            # self.cross_modal_ln_separate
+            self.ln_separate = nn.ModuleList([BridgeTowerLayerNorm(width) for _ in range(layers)])
+
+    def forward(self, x: torch.Tensor, x_mask):
+        visual_output = self.conv1(x)  # shape = [*, width, grid, grid]
+        visual_output = visual_output.reshape(
+            visual_output.shape[0], visual_output.shape[1], -1
+        )  # shape = [*, width, grid ** 2]
```

Comment:
nit: can you put the comment above so it doesn't force splitting across three lines? 
```suggestion
        # shape = [*, width, grid ** 2]
        visual_output = visual_output.reshape(visual_output.shape[0], visual_output.shape[1], -1)  
```

Reply:
Handled. Thanks!